backend/globaleaks/handlers/public.py

```diff
@@ -360,37 +327,35 @@ def _get_public_resources(store, language):
 
 
 class AhmiaDescriptionHandler(BaseHandler):
+    def _empt_gen(self):
+        return None
+
     @BaseHandler.transport_security_check("unauth")
     @BaseHandler.unauthenticated
     @inlineCallbacks
     def get(self):
         """
         Get the ahmia.fi descriptor
         """
-        node_info = yield GLApiCache.get('node', self.request.language,
-                                         serialize_node, self.request.language)
-
-        if node_info['ahmia']:
+        if GLSettings.memory_copy.ahmia:
             ret = yield GLApiCache.get('ahmia', self.request.language,
                                        serialize_ahmia, self.request.language)
 
             self.write(ret)
         else:  # in case of disabled option we return 404
+            yield self._empt_gen() # TODO remove inlineCallbacks from outer get.
```

Comment:
@NSkelsey: this can be simply replaced with "yield None" or even "yield" :)

Magic!